include/oneapi/dpl/optional

```diff
@@ -17,18 +17,18 @@
 #define __ONEDPL_optional
 
 #include "oneapi/dpl/internal/common_config.h"
+// ATTENTION!!! Include the header from the C++ standard library before the oneDPL config header to see whether the _PSTL_VERSION macro is defined
 #include <optional>
+#include "oneapi/dpl/pstl/onedpl_config.h"
```

Comment:
Potentially you discovered an interesting problem. On the other hand, it seems orthogonal to this review. Let's fix it separately.

Reply:
Fixed: this change was rolled back.